Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/DownsampleSinglePassMinAvgMax.azsl

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+// refer https://github.com/GPUOpen-Effects/FidelityFX-SPD/blob/c52944f547884774a1b33066f740e6bf89f927f5/sample/src/DX12/SPDIntegration.hlsl
+
+#include <Atom/Features/SrgSemantics.azsli>
+
+struct SpdGlobalAtomicBuffer
+{
+    uint m_counter;
+};
+
+ShaderResourceGroup PassSrg : SRG_PerPass
+{
+    RWTexture2D<float4> m_imageDestination[13]; // don't access MIP [6]. 
+    [[verbatim("globallycoherent")]] RWTexture2D<float4> m_imageDestination6;
+    [[verbatim("globallycoherent")]] RWStructuredBuffer<SpdGlobalAtomicBuffer> m_globalAtomic;
+
+    uint m_mips;
+    uint m_numWorkGroups;
+    uint2 m_workGroupOffset;
+}
+
+#define A_GPU
+#define A_HLSL
+#include <3rdParty/Features/PostProcessing/ffx-spd/ffx_a.h>
+
+groupshared AU1 s_spdCounter;
+
+#ifndef SPD_PACKED_ONLY
+
+groupshared AF1 s_spdIntermediateR[16][16];
+groupshared AF1 s_spdIntermediateG[16][16];
+groupshared AF1 s_spdIntermediateB[16][16];
+groupshared AF1 s_spdIntermediateA[16][16];
```

Comment:
For this particular shader, we don't need the 4th component in LDS right?

Reply:
Yes.  I deleted s_spdIntermediateA.